### ISSUE TITLE
Can I compute 2nd derivative of the loss computing from a fine-tuned model-w.r.t pre-trained model's input with higher

### ISSUE THREAD

**Issue by hkunzhe** [facebookresearch/higher#28]
Hi, I don't know much about meta-learning. But I'm faced with a problem similar to implement meta-update with Pytorch which is illustrated in [this Pytorch issue](https://github.com/pytorch/pytorch/issues/12659).

I have a pre-trained model W1 and two data point (x1, y1), (x2, y2). Firstly, I fine-tune W1 with (x1, y1) in one epoch, thus the model update ![771fe9804c97881f24d1ad69d0b1c81b0105018d](https://user-images.githubusercontent.com/30763967/73591435-5f1e0500-4529-11ea-8c2c-c49b288eb473.png) can be obtained by ![6f1f978c3cd6e99c2de916832b40867e37e745a4](https://user-images.githubusercontent.com/30763967/73591457-8d9be000-4529-11ea-9243-f3b4d60d257c.png)

And then, I get the test loss for (x2, y2) w.r.t fine-tuned model  ![a29f5e5a82cfb7a9d7dd43c65c19e4a7f7320f86](https://user-images.githubusercontent.com/30763967/73591478-d81d5c80-4529-11ea-88b9-deaf83436da0.png) by ![a35ba8298759e1ced1b7da62d12c69f1ac05a16f](https://user-images.githubusercontent.com/30763967/73591484-ef5c4a00-4529-11ea-84e8-d50924f94c25.png).

What I want to do is to use the test_loss to update x1 by doing
![bb47b40c6c79204038b44ba4e2ba34283bf389c3](https://user-images.githubusercontent.com/30763967/73591495-24689c80-452a-11ea-88e8-64e56b22d4c7.png)

Howerver, as in meta-learning, the Module parameters updates ![a29f5e5a82cfb7a9d7dd43c65c19e4a7f7320f86](https://user-images.githubusercontent.com/30763967/73591478-d81d5c80-4529-11ea-88b9-deaf83436da0.png) will perform in a non-differentiable way. So can I solve this problem with higher? Thank you in advance.

**Comment by egrefen** [comment]
Yes this is what the library is meant to help you do: in near-native pytorch (i.e. you can take existing code and make hopefully minimal changes) you can unroll the training loop of a model, and take gradient with regard to just about anything. The use-case you describe is just the basic use-case, so good news :)

Closing this issue, but please re-open if you need any clarification. Otherwise, I recommend looking at example code and experimenting yourself.
